src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -332,40 +340,18 @@ public SnowflakeStreamingIngestChannelInternal<?> openChannel(OpenChannelRequest
         getName());
 
     try {
-      Map<Object, Object> payload = new HashMap<>();
-      payload.put(
-          "request_id", this.flushService.getClientPrefix() + "_" + counter.getAndIncrement());
-      payload.put("channel", request.getChannelName());
-      payload.put("table", request.getTableName());
-      payload.put("database", request.getDBName());
-      payload.put("schema", request.getSchemaName());
-      payload.put("write_mode", Constants.WriteMode.CLOUD_STORAGE.name());
-      payload.put("role", this.role);
-      payload.put("is_iceberg", isIcebergMode);
-      if (request.isOffsetTokenProvided()) {
-        payload.put("offset_token", request.getOffsetToken());
-      }
-
-      OpenChannelResponse response =
-          executeWithRetries(
-              OpenChannelResponse.class,
-              OPEN_CHANNEL_ENDPOINT,
-              payload,
-              "open channel",
-              STREAMING_OPEN_CHANNEL,
-              httpClient,
-              requestBuilder);
-
-      // Check for Snowflake specific response code
-      if (response.getStatusCode() != RESPONSE_SUCCESS) {
-        logger.logDebug(
-            "Open channel request failed, channel={}, table={}, client={}, message={}",
-            request.getChannelName(),
-            request.getFullyQualifiedTableName(),
-            getName(),
-            response.getMessage());
-        throw new SFException(ErrorCode.OPEN_CHANNEL_FAILURE, response.getMessage());
-      }
+      OpenChannelRequestInternal openChannelRequest =
+          new OpenChannelRequestInternal(
```

Comment:
looks like you miss isOffsetTokenProvided?

Reply:
The `request.getOffsetToken()` is okay to be null. As `@JsonInclude(JsonInclude.Include.NON_NULL)` in `OpenChannelRequest` will ignore it automatically.

Reply:
It was used to avoid putting `null` value in payload string. As we are changing to json serialization now, using `@JsonInclude(JsonInclude.Include.NON_NULL)` will ignore the `null` fields. Removed `isOffsetTokenProvided` in `OpenChannelRequest`.